### ISSUE TITLE
PDF2 build fails on DITA-OT 3.7.2 with java.util.NoSuchElementException

### ISSUE THREAD

**Issue by c-kunz** [dita-ot/dita-ot#3970]
## Expected Behavior
PDF is build, replacing the missing glyph with a literal  `#` as did previous versions of DITA-OT (e.g. 3.6.1) and as described in https://xmlgraphics.apache.org/fop/2.6/fonts.html#missing-glyphs

## Actual Behavior
Build fails due to `java.util.NoSuchElementException`

## Possible Solution
The real solution is to
1. find glyphs that are available in the font to embed in PDF.
2. embed fonts that contain the glyphs used.

## Steps to Reproduce

Complete repository with steps to reproduce is available here:

https://github.com/c-kunz/failing-pdf2-dita-ot-build

## Copy of the error message, log file or stack trace
```
      [fop] [ERROR] Anttask - Error rendering fo file: /tmp/temp20220721110741020/topic.fo <org.apache.fop.apps.FOPException: java.util.NoSuchElementException
      [fop] java.util.NoSuchElementException>org.apache.fop.apps.FOPException: java.util.NoSuchElementException
      [fop] java.util.NoSuchElementException
      [fop] 	at org.apache.fop.tools.anttasks.FOPTaskStarter.renderInputHandler(Fop.java:652)
      [fop] 	at org.apache.fop.tools.anttasks.FOPTaskStarter.render(Fop.java:671)
      ...
      (See full log in https://github.com/c-kunz/failing-pdf2-dita-ot-build )
```

## Environment
* DITA-OT version: 3.7.2
* Operating system and version: Linux Ubuntu 20.04
* How did you run DITA-OT? `dita -f pdf2 -i repro.dita`
* Transformation type: PDF2

## Hints
 * Works as expected on DITA-OT 3.6.1
 * I could'nt reproduce this issue with a standalone file and FOP 2.6 (the missing glyph is replaced correctly with `#`)

**Comment by raducoravu** [comment]
@c-kunz the Apache FOP processor has a ".fop" folder located in your user home folder containing cached fonts and various other caches, can you remove that ".fop" folder from your side and try to publish again?

**Comment by c-kunz** [comment]
Sure, I just ran the following commands:

```bash
rm -rf ~/.fop
dita -f pdf2 -i repro.dita --debug
```

But the same issue occurs.

Since I couldn't reproduce this behavior with DITA-OT 3.6.1 or with a standalone FOP 2.6 I have assumed that the problem must be located somewhere in between.

**Comment by raducoravu** [comment]
I'm afraid I do not have Ubuntu, I tested on Mac OS and one of my colleagues tested on Windows and we cannot reproduce the problem. This might also be related to the fonts you have installed on your computer.

**Comment by c-kunz** [comment]
Thanks for the tip @raducoravu !

I've tested this again on a different Windows 10 machine and on Ubuntu 22.04 but failed to reproduce on either of those. Afterwards, I simply installed a font that contained the missing glyphs system-wide on Ubuntu 20.04, and the build works again.

The only thing that baffles me is that DITA-OT 3.6.x works on that machine and replaces missing glyphs with `#` like FOP usually does, but fails to build on 3.7.x

Anyhow, I would consider this issue as resolved. Thanks! :-)
